packages/backend/src/managers/catalogManager.spec.ts

```diff
@@ -305,27 +289,37 @@ test('malformed catalog should create a notification', async () => {
   );
 });
 
-test('catalog with undefined version should call sanitize function to try converting it', () => {
-  const sanitizeSpy = vi.spyOn(catalogUtils, 'sanitize');
-  const writeFileMock = vi.spyOn(promises, 'writeFile').mockResolvedValue();
-  catalogManager['onUserCatalogUpdate']({
-    recipes: [
-      {
-        id: 'chatbot',
-        description: 'This is a Streamlit chat demo application.',
-        name: 'ChatBot',
-        repository: 'https://github.com/containers/ai-lab-recipes',
-        ref: 'v1.1.3',
-        icon: 'natural-language-processing',
-        categories: ['natural-language-processing'],
-        basedir: 'recipes/natural_language_processing/chatbot',
-        readme: '',
-        models: ['hf.instructlab.granite-7b-lab-GGUF', 'hf.instructlab.merlinite-7b-lab-GGUF'],
-      },
-    ],
-    models: [],
+describe('spy on catalogUtils.sanitize', () => {
+  beforeEach(() => {
+    // do not mock catalogUtils, as we want to use the original catalogUtils.hasCatalogWrongFormat
+    vi.spyOn(catalogUtils, 'sanitize');
```

Comment:
question: not sure to understand the comment and the line after the comment ?

Reply:
I updated the comment